src/macros.rs

```diff
@@ -715,6 +715,82 @@ macro_rules! query_file_scalar_unchecked (
+/// one ore more source files have changed, because normally it shouldn't have to otherwise. SQLx is
+/// just weird in that external factors can change the output of proc macros, much to the chagrin of
+/// the compiler team and IDE plugin authors.
+///
+/// As of 0.5.6, we emit `include_str!()` with an absolute path for each migration, but that
+/// only works to get the compiler to watch _existing_ migration files for changes.
+///
+/// Our only options for telling it to watch the whole `migrations/` directory are either via the
+/// user creating a Cargo build script in their project, or using an unstable API on nightly
+/// governed by a `cfg`-flag.
+///
+/// ##### Stable Rust: Cargo Build Script
+/// The only solution on stable Rust right now is to create a Cargo build script in your project
+/// and have it print `cargo:rerun-if-changed=migrations`:
+///
+/// `build.rs`
+/// ```
+/// fn main() {
+///     println!("cargo:rerun-if-changed=migrations");
+/// }
+/// ```
+///
+/// You can also do this automatically by running `sqlx migrate build-script`.
+///
+/// See: [The Cargo Book: 3.8 Build Scripts; Outputs of the Build Script](https://doc.rust-lang.org/stable/cargo/reference/build-scripts.html#outputs-of-the-build-script)
+///
+/// #### Nightly Rust: `cfg` Flag
+/// The `migrate!()` macro also listens to `--cfg procmacro2_semver_exempt`, which will enable
+/// the `track_path` feature to directly tell the compiler to watch the `migrations/` directory:
+///
+/// ```sh,ignore
+/// $ env RUSTFLAGS='--cfg procmacro2_semver_exempt' cargo build
+/// ```
+///
+/// Note that this unfortunately will trigger a fully recompile of your dependency tree, at least
+/// for the first time you use it.
+///
+/// You can also set it in `build.rustflags` in `.cargo/config.toml`:
+/// ```toml,ignore
+/// [build]
+/// rustflags = ["--cfg procmacro2_semver_exempt"]
+/// ```
+///
+/// And then continue building and running your project normally.
+///
+/// If you're building on nightly anyways, it would be extremely helpful to help us test
+/// this feature and find any bugs in it.
+///
+/// Subscribe to [the `track_path` tracking issue](https://github.com/rust-lang/rust/issues/73921)
+/// for discussion and the future stabilization of this feature.
+///
+/// **Why not a feature flag?** Feature flags enabling nightly features are technically an anti-pattern
+/// because they take away agency from the person compiling the final binary, even though
+/// the choice of stable vs nightly _should_ be up to them. If you're using SQLx in a library, you
+/// should not set this flag but instead direct your end-users to use it, if applicable.
+///
+/// Additionally, enabling the `nightly` feature would subject you to possible breaking changes
+/// between nightly versions of Rust, even though the version of SQLx hasn't changed. By
+/// forcing the use of the `cfg`-flag, you are required to acknowledge the potential for
+/// breaking changes.
```

Comment:
To me, prefixing a cargo feature with `unstable-` or `semver-exempt-` conveys the same thing.